packages/yarnpkg-core/sources/Project.ts

```diff
@@ -812,6 +827,13 @@ export class Project {
         if (typeof finalResolution === `undefined`)
           throw new Error(`${structUtils.prettyDescriptor(this.configuration, descriptor)}: No candidates found`);
 
+        if (opts.checkResolutions) {
+          const {locators} = await noLockfileResolver.getSatisfying(descriptor, resolvedDependencies, [finalResolution], {...resolveOptions, resolver: noLockfileResolver});
+          if (!locators.find(locator => locator.locatorHash === finalResolution.locatorHash)) {
+            throw new ReportError(MessageName.RESOLUTION_MISMATCH, `Invalid resolution ${structUtils.prettyResolution(this.configuration, descriptor, finalResolution)}`);
```

Comment:
I feel like the error message could include a more detailed explanation detailing what it means that the resolution is invalid, how it might've happened and how it could be fixed.

Reply:
I'd like to keep messages on one line if possible, so I was thinking to explain that in more details within the website's error page.

Reply:
:thinking: We can do it like that for now, but in the future, I feel like actually showing helpful advice directly (without having to navigate to the website) would be an important UX improvement. I really like the way you improved the project detection error message for example.